src/arch/aarch64.rs

```diff
@@ -0,0 +1,229 @@
+        } else {
+            align_to(buffer, get_integer_layout(8).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for u16 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&(*self as u64).to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(16).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i16 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&(*self as u64).to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(16).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for u32 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&(*self as u64).to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(32).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i32 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&(*self as u64).to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(32).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for u64 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(64).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i64 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(64).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for u128 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(128).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i128 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(128).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for Felt {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
```

Comment:
can you add a small comment here why it checks against 56? I think it can improve readability

Reply:
Done. I added at the module level because the `64` and `56` numbers are used in various places.